src/server/scripts/KulTiras/ShrineOfTheStorm/boss_tidesage_council.cpp

```diff
@@ -0,0 +1,541 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AreaTrigger.h"
+#include "AreaTriggerAI.h"
+#include "Creature.h"
+#include "GridNotifiers.h"
+#include "InstanceScript.h"
+#include "Map.h"
+#include "ObjectAccessor.h"
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "SpellAuraEffects.h"
+#include "SpellScript.h"
+#include "TaskScheduler.h"
+#include "shrine_of_the_storm.h"
+
+enum TidesageCouncilSpells
+{
+    // Brother Ironhull
+    SPELL_BLESSING_OF_IRONSIDES                = 267901,
+    SPELL_HINDERING_CLEAVE                     = 267899,
+    SPELL_REINFORCING_WARD_AREATRIGGER         = 267905,
+    SPELL_REINFORCING_WARD_DAMAGE_TAKEN_PLAYER = 267904, // Pets can get buff too
+    SPELL_REINFORCING_WARD_DAMAGE_TAKEN_BOSS   = 268186,
+    SPELL_REINFORCING_WARD_DISPEL_BOSS         = 272001,
+    SPELL_REINFORCING_WARD_DISPEL_PLAYER       = 272002,
+    SPELL_REINFORCING_WARD_SELECTOR            = 267903,
+    SPELL_COSMETIC_REINFORCING_WARD            = 274690,
+    SPELL_BROTHER_IRONHULL_GAIN_ENERGY         = 267916, // Serverside
+
+    // Galecaller Faye
+    SPELL_BLESSING_OF_THE_TEMPEST              = 267830,
+    SPELL_SLICING_BLAST                        = 267818,
+    SPELL_BLOWBACK_MISSILE                     = 267842,
+    SPELL_BLOWBACK_DAMAGE                      = 267838,
+    SPELL_BLOWBACK_AREATRIGGER                 = 267836,
+    SPELL_SWIFTNESS_WARD_SELECTOR              = 267902,
+    SPELL_SWIFTNESS_WARD_AREATRIGGER           = 267891,
+    SPELL_SWIFTNESS_WARD_HASTE_BOSS            = 267890,
+    SPELL_SWIFTNESS_WARD_HASTE_PLAYER          = 267888,  // Pets can get buff too
+    SPELL_COSMETIC_SWIFTNESS_WARD              = 274689,
+    SPELL_GALECALLER_FAYE_GAIN_ENERGY          = 267894 // Serverside
+};
+
+enum TidesageCouncilEvents
+{
+    EVENT_BLESSING_OF_IRONSIDES = 1,
+    EVENT_HINDERING_CLEAVE,
+    EVENT_REINFORCING_WARD,
+
+    EVENT_BLESSING_OF_THE_TEMPEST,
+    EVENT_SLICING_BLAST,
+    EVENT_SWIFTNESS_WARD,
+
+    EVENT_CHECK_MANA
+};
+
+enum TidesageCouncilTexts
+{
+    SAY_AGGRO                    = 0,
```

Comment:
this is minimal but i would use 0-4 for generic ones, and then 5-X for specific ones to keep it ordered